pkg/test/e2e/ccm-migration/cluster.go

```diff
@@ -211,7 +217,84 @@ func (c *ClusterJig) createCluster(cloudSpec kubermaticv1.CloudSpec) error {
 
 // CleanUp deletes the cluster.
 func (c *ClusterJig) CleanUp() error {
-	return c.SeedClient.Delete(context.TODO(), c.Cluster)
+	clusterClientProvider, err := clusterclient.NewExternal(c.SeedClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to get user cluster client provider")
+	}
+
+	userClient, err := clusterClientProvider.GetClient(context.TODO(), c.Cluster)
```

Comment:
Please put the context into a `ctx` variable, we use tons of TODO()s in this function.